zcash_client_backend/src/data_api.rs

```diff
@@ -1565,10 +1565,42 @@ impl AccountBirthday {
 
 /// This trait encapsulates the write capabilities required to update stored wallet data.
 ///
+/// # Adding accounts
+///
+/// This trait provides several methods for adding accounts to the wallet data:
+/// - [`WalletWrite::create_account`]
+/// - [`WalletWrite::import_account_hd`]
+/// - [`WalletWrite::import_account_ufvk`]
+///
+/// All of these methods take an [`AccountBirthday`]. The birthday height is defined as
+/// the minimum block height that will be scanned for funds belonging to the wallet. If
+/// `birthday.height()` is below the current chain tip, the account addition operation
+/// will trigger a re-scan of the blocks at and above the provided height.
+///
+/// The order in which you call these methods will affect the resulting wallet structure:
+/// - If only [`WalletWrite::create_account`] is used, the resulting accounts will have
+///   sequential [ZIP 32] account indices within each given seed.
+/// - If [`WalletWrite::import_account_hd`] is used to import accounts with non-sequential
+///   ZIP 32 account indices from the same seed, a call to [`WalletWrite::create_account`]
+///   will use the ZIP 32 account index just after the highest-numbered existing account.
+/// - If an account is imported via [`WalletWrite::import_account_ufvk`], and then a later
+///   call to either [`WalletWrite::create_account`] or [`WalletWrite::import_account_hd`]
+///   would produce the same UFVK, an error will be returned.
```

Comment:
Should this say "If an account created via [`WalletWrite::create_account`] or imported via [`WalletWrite::import_account_ufvk`] or [`WalletWrite::import_account_hd`] already exists in the wallet with the same UFVK (which will also be the case for accounts with the same seed), an error will be returned." ?

This differs from the current wording because (I think) it is not the case that the creation or import has to be in any particular order; any of these three methods will fail if it produces a duplicate UFVK, regardless of which of them was used to create the existing account. At least, if that's not the case then it is a bug.